src/mol-plugin-state/manager/structure/selection.ts

```diff
@@ -484,6 +491,31 @@ export class StructureSelectionManager extends StatefulPluginComponent<Structure
         }
     }
 
+    getSnapshot(): StructureSelectionSnapshot {
+        const entries: StructureSelectionSnapshot['entries'] = [];
+
+        this.entries.forEach((entry, ref) => {
+            entries.push({
+                ref,
+                bundle: StructureElement.Bundle.fromLoci(entry.selection)
```

Comment:
Could it happen that the bundle is using a typed array? In that case should convert it to native array.

Reply:
We don't prevent it with type checking but the `Bundle.from*` methods don't use typed arrays. Got an idea to enforce that easily? We already include `Bundle`s in the state in `StructureSelectionFromBundle`, `OverpaintStructureRepresentation3DFromBundle` and other transforms.

Reply:
Only to traverse the tree. But yeah, been using it in the state tree already, so should be fine.